geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -381,6 +381,55 @@ long zrem(Region<RedisKey, RedisData> region, RedisKey key, List<byte[]> members
     return getRange(min, max, withScores, true);
   }
 
+  List<byte[]> zrevrangebyscore(SortedSetScoreRangeOptions rangeOptions, boolean withScores) {
+    AbstractOrderedSetEntry maxEntry =
+        new ScoreDummyOrderedSetEntry(rangeOptions.getStartRange(), rangeOptions.isStartExclusive(),
+            false);
+    int maxIndex = scoreSet.indexOf(maxEntry);
+
+    AbstractOrderedSetEntry minEntry =
+        new ScoreDummyOrderedSetEntry(rangeOptions.getEndRange(), rangeOptions.isEndExclusive(),
+            true);
+    int minIndex = scoreSet.indexOf(minEntry);
+    if (minIndex > getSortedSetSize()) {
+      return Collections.emptyList();
+    }
+
+    if (minIndex == maxIndex) {
+      return Collections.emptyList();
+    }
+
+    // Okay, if we make it this far there's a potential range of things to return.
+    int count = Integer.MAX_VALUE;
+    if (rangeOptions.hasLimit()) {
+      count = rangeOptions.getCount();
+      maxIndex -= rangeOptions.getOffset();
+      if (maxIndex < 0) {
+        return Collections.emptyList();
+      }
+    }
+
+    int startIndex = maxIndex - 1;
+    int endIndex = Math.min(count, maxIndex - minIndex);
+    int initialCapacity = startIndex - endIndex;
+    if (withScores) {
+      initialCapacity *= 2;
+    }
+
+    Iterator<AbstractOrderedSetEntry> entryIterator =
+        scoreSet.getIndexRange(startIndex, endIndex, true);
+    List<byte[]> result = new ArrayList<>(initialCapacity > 0 ? initialCapacity : 0);
```

Comment:
if initialCapacity is <= 0 does that mean we will return an empty list? If so should we just have checked (up at line 414) if (initialCapacity <= 0) {return Collections.emptyList();}? Then this could just be new ArrayList<>(initialCapacity)

Reply:
Yeah, per Donal we're using the addLimitToRange (with some tweaks now).